crates/kitsune_p2p/kitsune_p2p/src/gossip/sharded_gossip/tests/sharded_scenario.rs

```diff
@@ -0,0 +1,123 @@
+use kitsune_p2p_types::dht_arc::{ArcInterval, DhtLocation};
+use maplit::hashset;
+use std::collections::{BTreeSet, HashSet};
+
+/// Abstract representation of the instantaneous state of a sharded network
+/// with multiple conductors. Useful for setting up multi-node test scenarios,
+/// and for deriving the expected final state after reaching consistency.
+///
+/// NB: The "reification" of this representation will break a rule:
+/// Agent and Op hashes will have manually defined locations which do NOT
+/// match the actual hash content, since it is computationally infeasible to
+/// search for hashes which match a given location.
+/// (By "reification" I mean the actual concrete Nodes which are set up.)
+///
+/// Thus, rather than dealing with hash types directly, this representation
+/// deals only with locations.
+///
+/// Thus, note that for simplicity's sake, it's impossible to specify two ops
+/// at the same location, which is possible in reality, but rare, and should
+/// have no bearing on test results. (TODO: test this case separately)
```

Comment:
Yeah, I think it may be worth the simplicity to keep this layer of indirection out, but if we wind up needing it later, even if we've already written a lot of tests using this way, we can write a more general struct with IDs, and impl From this to that.